Updates to the Developer Guide

sarina opened the pull request:
Updates some small typos, adds a section on GitHub Actions, and removes some outdated references related to testing and XBlocks

sarina commented:
@feanil is there someone you can think of to review this that would know about testing & github actions?

feanil commented:
Someone on arch-bom/arbi-bom at 2U might be a good candidate for that.  @jmbowman do you have someone that you could recommend?

sarina commented:
@feanil it's been 5 months but I have addressed Jeremy's one comment and got the build working again. \o/

I'll tag @brian-smith-tcril - Brian, you and I fixed up a good portion of the dev guide together. Any desire to take a run through this?

brian-smith-tcril commented on source/developers/references/developer_guide/testing/github-actions.rst:
I'm having a hard time understanding the purpose of this file. It seems to be serving as an introduction to GitHub Actions as a concept, an overview of how GitHub Actions are used throughout the Open edX GH org, but also a couple how-tos for troubleshooting specific checks (GH Actions and otherwise) that exist on most repos.

I feel like we'd benefit from having:
* What are GitHub Actions?
* How are GitHub  Actions used in the Open edX project?
* How to troubleshoot? automated checks on repos in the Open edX project.

sarina replied:
I restructured the document to have these three "sections" in this doc. I think it's a good framing.

brian-smith-tcril reviewed (commented):
Left a few initial thoughts! Please let me know if you have any questions for me!
